modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/CreateAnAPIApplicationEndpoint.testcase

```diff
@@ -93,6 +139,346 @@ definition {
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "path-defaul");
+
+			Type(
+				locator1 = "APIBuilder#ENDPOINT_DESCRIPTION",
+				value1 = "Description default");
+
+			Click(locator1 = "APIBuilder#CREATE_BUTTON");
+		}
+
+		task ("Then I can see success alert") {
+			Alert.viewSuccessMessage();
+		}
+
+		task ("And Then I can see info of created endpoint with 'Get' method, scope, path") {
+			AssertElementPresent(
+				key_optionName = "GET",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+
+			AssertElementPresent(
+				key_optionName = "Company",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+
+			AssertElementPresent(
+				key_fieldLabel = "Path",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "path-defaul");
+
+			AssertElementPresent(
+				locator1 = "APIBuilder#ENDPOINT_DESCRIPTION",
+				value1 = "Description default");
+		}
+	}
+
+	@priority = 4
+	test CanSeePathOfPathFields {
+		property portal.acceptance = "true";
+
+		task ("Given the API Application is created") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "test2",
+				title = "test2");
+		}
+
+		task ("When I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I click 'Add API Endpoint' button") {
+			Click(locator1 = "Button#PLUS");
+		}
+
+		task ("Then I can see path under {portalUrl}/o/c/{applicationName}") {
+			AssertTextEquals(
+				locator1 = "APIBuilder#URL_LABEL",
+				value1 = "//localhost:8080/o/c/title2");
+		}
+
+		task ("And Then I can see path start with '/' with placeholder 'Enter path'") {
+			AssertElementPresent(locator1 = "APIBuilder#SLASH");
+		}
+	}
+
+	@priority = 4
+	test CanSeeValidationMessageForRequiredFields {
+		property portal.acceptance = "true";
+
+		task ("Given the API Application is created") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "test2",
+				title = "test2");
+		}
+
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I click 'Add API Endpoint' button") {
+			Click(locator1 = "Button#PLUS");
+		}
+
+		task ("When I click 'Create' button in 'New API Endpoint'") {
+			Click(locator1 = "APIBuilder#CREATE_BUTTON");
+		}
+
+		task ("Then I can see error message about required Scope and Path") {
+			AssertElementPresent(
+				key_message_error = "Please select a scope",
+				locator1 = "APIBuilder#MESSAGE_ERROR");
+
+			AssertElementPresent(
+				key_message_error = "Please enter a path",
+				locator1 = "APIBuilder#MESSAGE_ERROR");
+		}
+
+		task ("And Then no endpoint is being created") {
+			Click(locator1 = "CP2Utils#CLOSE_BUTTON");
+
+			AssertElementPresent(locator1 = "APIBuilder#NO_API_ENDPOINT_FOUND");
+		}
+	}
+
+	@priority = 3
+	test CanValidateMaximumDescription {
```

Comment:
the task was to create only FP4 and FP5 tests 😉

Reply:
Hey @magjed4289 I put it following the information in the[ ticket](https://liferay.atlassian.net/browse/LPS-197668), but thanks for reporting this, could you tell me if in this case it would be 4 or 5, or teach me how I could analyze it myself? What are the parameters and guidelines for this?

Reply:
Ok, Done!